core/src/data_sources/folder.rs

```diff
@@ -1,8 +1,91 @@
 use serde::{Deserialize, Serialize};
 
-#[derive(Debug, Clone, Serialize, PartialEq, Deserialize)]
+use crate::project::Project;
+
+use super::node::{Node, NodeType, ToNode};
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
 pub struct Folder {
-    pub data_source_id: String,
-    pub folder_id: String,
-    pub created: u64,
+    project: Project,
+    data_source_id: String,
+    folder_id: String,
+    created: u64,
+    timestamp: u64,
+    title: String,
+    parents: Vec<String>,
+}
+
+/// MIME type used to identify folder objects
+pub const FOLDER_MIMETYPE: &str = "application/vnd.dust.folder";
+
+impl Folder {
+    pub fn new(
+        project: &Project,
+        data_source_id: &str,
+        folder_id: &str,
+        created: u64,
+        timestamp: u64,
+        title: &str,
+        parents: Vec<String>,
+    ) -> Self {
+        Folder {
+            project: project.clone(),
+            data_source_id: data_source_id.to_string(),
+            folder_id: folder_id.to_string(),
+            created,
+            timestamp,
+            title: title.to_string(),
+            parents,
+        }
+    }
+
+    pub fn from_node(node: &Node) -> Self {
+        Folder::new(
+            node.project(),
+            node.data_source_id(),
+            node.node_id(),
+            node.created(),
```

Comment:
This one is a bit of a lie isn't it assuming Folders and Nodes have a created?

What's the thinking around `created` for Node/Folder/DataSource ? Maybe we remove created from Node?

Reply:
Ah I see you removed created from folders, but what about document and data source?

I feel like created is best on the raw object vs node?

Reply:
Could be on both but then we'll need some gymnastic.